llvm/lib/MC/MCObjectStreamer.cpp

```diff
@@ -259,13 +259,30 @@ void MCObjectStreamer::emitLabelAtPos(MCSymbol *Symbol, SMLoc Loc,
   Symbol->setOffset(Offset);
 }
 
-void MCObjectStreamer::emitULEB128Value(const MCExpr *Value) {
+void MCObjectStreamer::emitULEB128Value(const MCExpr *Value,
+                                        unsigned PadTo) {
   int64_t IntValue;
+  // Avoid fixups when possible.
   if (Value->evaluateAsAbsolute(IntValue, getAssemblerPtr())) {
     emitULEB128IntValue(IntValue);
     return;
   }
-  insert(getContext().allocFragment<MCLEBFragment>(*Value, false));
+
+  if (!PadTo) {
```

Comment:
To be even more specific here, the thing I was hoping for was that we could tell, based on some property of `Value`, whether there needs to be a relocation. If so, we can basically just assume that the size will be 5 (The only non-32-bit relocation I know of would be for memory addresses, so there could be i64.const instruction in code with 9-byte relocatable LEBs, but I guess those are handled differently elsewhere). And then we don't need the explicit PadTo in the streamer API or the separate assembler directive. 
The question of course, is whether that's possible. MCExpr has `evaluateAsRelocatable` alongside `evaluateAsAbsolute` but currently I'm not totally sure what it means.